CONTRIBUTING.md

```diff
@@ -11,95 +11,156 @@ open an issue and discuss the feature with us. Sending a PR without discussion
 might result in a rejected PR, because we might be taking the repository in a
 different direction.
 
-## Building from source
+## Setting up the Workspace
 
-We recommend you use our prebuilt Docker image to start your development work
-using either VS Code or a local container:
+Please follow the steps below in order:
 
-### Visual Studio Code Dev Container
+### Prerequisite
 
-* Create an empty directory for your workspace on your development host. These
-  instructions assume you are using a remote host and are connecting to it over
-  SSH.
+To work on PyTorch/XLA, you'll need a powerful Linux machine with plenty of
+CPUs and RAM. Make sure you have `git` and `docker` installed on this machine.
+If you don't, follow https://github.com/git-guides/install-git and
+https://docs.docker.com/engine/install/ to install them.
 
-* Clone PyTorch, TorchVision, and PyTorch/XLA into your workspace directory:
+### Forking the Git Repos
 
-```bash
-  git clone --recursive --depth=1 https://github.com/pytorch/pytorch.git
+In order to create PRs later, we need to first fork the Git repos we'll be
+working with:
 
-  # Install TorchVision if you need to run tests that involve vision modules
-  git clone --recursive --depth=1 https://github.com/pytorch/vision.git
+1.  Go to https://github.com/pytorch/pytorch and fork it as `pytorch`.
```

Comment:
you only need to fork torch and torchvision if you plan to make changes to it (and upstream).

Most of cases we don't need to do that.

Reply:
If one cloned pytorch/pytorch thinking there is no need to make changes upstream, and later they want to contribute to pytorch/pytorch, you can still fork then and add an upstream with git remote add. So we can keep the default path cloning directly pytorch/pytorch

Reply:
Thanks for the advice. My thinking is to minimize the number of configurations we support. If some people fork pytorch and some don't, our instructions will have to cover for both cases, which can add more burden to both the readers and the maintainers. In the same spirit, we decided that internally we'll only support GCE VMs, even though it might not be strictly necessary. Let me know if this makes sense!